mariadb-connector/src/main/java/life/qbic/persistence/OffsetBasedRequest.java

```diff
@@ -52,11 +66,14 @@ public long getOffset() {
   }
 
   @Override
+  @NonNull
   public Sort getSort() {
+    Objects.requireNonNull(sort);
```

Comment:
You could also think about moving the @Nonnull annotation to the sort field value with a description such as:
 @NotNull(message = "Sorting may not be null")

Reply:
TBH. the sort can never be null as it is set in all constructors, so I would not expect a null pointer here 😄 
I added the annotations because IntelliJ hinted me to the fact that they are expected by the interface method.